modules/consent/lib/Auth/Process/Consent.php

```diff
@@ -144,13 +144,40 @@ public function __construct($config, $reserved)
     /**
      * Helper function to check whether consent is disabled.
      *
-     * @param mixed $option  The consent.disable option. Either an array or a boolean.
+     * @param mixed $option  The consent.disable option. Either an array of array, an array or a boolean.
      * @param string $entityIdD  The entityID of the SP/IdP.
      * @return boolean  TRUE if disabled, FALSE if not.
      */
     private static function checkDisable($option, $entityId) {
         if (is_array($option)) {
-            return in_array($entityId, $option, TRUE);
+            // Check if consent.disable array has one element that is an array
+            if (count($option) === count($option, COUNT_RECURSIVE)) {
+                // Array is not multidimensional.  Simple in_array search suffices
+                return in_array($entityId, $option, true);
+            } else {
```

Comment:
Given that the main `if` clause will always end the execution of this method, the `else` is useless here, it only adds an indentation level and increases complexity. You can just remove the `else` and have the code inside execute next to the `if`:

```php
if (count($option) === count($option, COUNT_RECURSIVE)) {
    // Array is not multidimensional.  Simple in_array search suffices
    return in_array($entityId, $option, true);
 }

// Array contains at least one element that is an array, verify both possibilities
if (in_array($entityId, $option, true)) {
    return true;
}
```